metrics-core/src/main/java/software/amazon/swage/metrics/record/cloudwatch/CloudWatchRecorder.java

```diff
@@ -279,7 +279,7 @@ protected void record(
             final Number value,
             final Unit unit,
             final Instant time,
-            final TypedMap context)
+            final MetricContext context)
```

Comment:
By pulling the MetricContext out separate from a recorder instance, you have this weird case where the context containing a recorder is sent as parameter to a recorder...  should these actually be changed, or should they still take the metadata/dimensions directly?

Previously the Context was simply a convenience pairing the actual context (TypedMap) with the recorder for that context.  Now it's not quite so clean.

Reply:
Adding information on the parent/child relationship to the user's data did not seem right to me; nor did flattening the map. We could separate framework data into another class that we can pass back, but now we're adding layers we don't really need given the Context/Recorder contract is protected and we control Context's implementation.

Reply:
Thinking about this a bit more, it's probably worth looking at having an interface that application code can use to emit metrics that is totally separate from the context management and recorder management aspects.